### PR TITLE
*FileLoader -> *FileOpener

### PR DESCRIPTION
**Pull request by pmeier**
Addresses pytorch/pytorch#70367 and pytorch/data#148. CI failures are expected until the other two PRs are landed.

cc @pmeier @bjuncek

### CODE REVIEW

**Review by NivekT — APPROVED** [review]
LGTM! Thanks for making these changes. Please wait until the other 2 PRs land to land this one. I will let you know.

File: torchvision/prototype/datasets/_folder.py
Comment (NivekT) [inline torchvision/prototype/datasets/_folder.py:57]: @pmeier I should mention that `FileLoader` has a default `mode="r"`, and `FileOpener` has a default `mode="b"`. We made that change to be consistent with other libraries. You may also want to specify the modes here as it fits your use cases.

File: torchvision/prototype/datasets/utils/_resource.py
Comment (NivekT) [inline torchvision/prototype/datasets/utils/_resource.py:65]: Default is now `mode="b"`, not `mode="r"`.

File: torchvision/prototype/datasets/utils/_resource.py
Comment (NivekT) [inline torchvision/prototype/datasets/utils/_resource.py:67]: Default is now `mode="b"`, not `mode="r"`.

File: torchvision/prototype/datasets/_folder.py
Comment (pmeier) [reply torchvision/prototype/datasets/_folder.py]: You have these mixed up, right? `FileLoader` had `mode="b"`, but `FileOpener` will now go with `mode="r"`, correct?

Comment (NivekT) [reply torchvision/prototype/datasets/_folder.py]: Yep you're right. Sorry

**Comment by NivekT** [comment]
@pmeier I have merged the TorchData PR. The CI tests here mostly look good, but there are a few failures (seem unrelated to me).

**Pull request merged**
